src/pages/user.tsx

```diff
@@ -26,20 +28,49 @@ const userData = [
     projects: ["Create your Burger", "Learning and Laughing"],
     imageUrl: "https://i.imgur.com/knDwrmG.jpg",
   },
+  {
+    id: 3,
+    name: "Bill Gates",
+    birthDate: "705888000000",
+    position: "Designer UI/UX",
+    salary: 4000,
+    projects: ["My Life Dashboard"],
+    imageUrl: "https://i.imgur.com/v76rx7g.jpg",
+  },
 ];
 
 export function User() {
   const [showModal, setShowModal] = useState(false);
+  const [data, setData] = useState<UserInterface[]>(userData);
 
   const openModal = () => {
     setShowModal(true);
   };
 
+  const handleSearchTermChange = (term: string) => {
+    if (term.length > 0) {
+      const filteredUsers = data.filter((user) => {
+        const lowerCaseTerm = term.toLowerCase();
+        return (
+          user.name.toLowerCase().includes(lowerCaseTerm) ||
+          user.position?.toLowerCase().includes(lowerCaseTerm) ||
+          user.projects?.some((project) =>
+            project.toLowerCase().includes(lowerCaseTerm)
+          )
+        );
+      });
+      setData(filteredUsers);
+    } else {
+      setData(userData);
```

Comment:
I suggest you run this logic to save the data inside a useMemo variable, instead of setting the state. For the example we have here, it's fine and working, because you have the original data always there, hardcoded and intact. But by the time you start integrating with the BE, if you lose the original data fetched, it might be hard to reset it to the original state.
So my suggestion is: keep the logic (that's good by the way!), but instead of setting the state to save the filtered data, put it into a useMemo, so in this way, you don't modify the original data and it'll make it easier for you to reset to the original form.

Reply:
Great, i switch to create the filteredUsers as a const using useMemo hook